### PR TITLE
Eliminate contentChanged parameter

### PR DESCRIPTION
**Pull request by tmat**
Calculating the value is simple and more reliable. It will also allow us to call the APIs when we do not know what exact changes were made.

### CODE REVIEW

**Comment by tmat** [comment]
@ToddGrun ptal

File: src/Workspaces/Core/Portable/Workspace/Solution/ProjectState.cs [hunk 1]
```diff
@@ -923,13 +921,16 @@ public ProjectState UpdateDocumentsOrder(ImmutableList<DocumentId> documentIds)
...
-        bool contentChanged,
         out AsyncLazy<VersionStamp> dependentDocumentVersion,
         out AsyncLazy<VersionStamp> dependentSemanticVersion)
     {
         var recalculateDocumentVersion = false;
         var recalculateSemanticVersion = false;
 
+        var contentChanged = !oldDocuments.SequenceEqual(
+            newDocuments,
+            static (oldDocument, newDocument) => oldDocument.TextAndVersionSource == newDocument.TextAndVersionSource);
```
Comment (ToddGrun) [inline src/Workspaces/Core/Portable/Workspace/Solution/ProjectState.cs:932]: >oldDocument.TextAndVersionSource == newDocument.TextAndVersionSource [](http://example.com/codeflow?start=49&length=68)

This seems a bit odd. @CyrusNajmabadi -- is this an ok comparison?

Comment (CyrusNajmabadi) [reply src/Workspaces/Core/Portable/Workspace/Solution/ProjectState.cs:932]: don't know :)  doens't seem like a terrible approach though.   if they're thes ame, they have the same content.  if they're differernt, they're likely to be different.

Comment (ToddGrun) [reply src/Workspaces/Core/Portable/Workspace/Solution/ProjectState.cs:932]: OK, then this looks good. I wasn't sure about the "likely to be different" part of your statement

**Review by ToddGrun — APPROVED** [review]
:shipit:

**Pull request merged**